### ISSUE TITLE
Admission structures not in public headers

### ISSUE THREAD

**Issue by wolftobias** [openssl/openssl#3780]
Today I tried to build the master branch to test the new admission feature. But unfortunately I found out that the admission structures are not in the public area available yet. Please make the admission header public.

v3_admis.h

**Comment by FdaSilvaYY** [comment]
Structures definitions can be straightforward moved to public headers, because there are now Opaque !
Only a bunch of accessor-methods will be added to public headers.
And so you will have to rewrite your code to call them, instead of accessing directly structures fields.

**Comment by wolftobias** [comment]
Yes I know and I understand what your are talking to me, I don`t want to use the structure directly. Sorry for the misunderstanding. What I mean is that even the pointers are not available.

e.g.
```
ADMISSION_SYNTAX* admission_syntax;
```

There are no admission stuff anyway in the public include files.

**Comment by richsalz** [comment]
Well, except maybe ASN1 structures which are defined by standards.

**Comment by wolftobias** [comment]
I was searching also for ADMISSION_SYNTAX_free function which is also not available. That is a standard function, isn`t it?!

**Comment by richsalz** [comment]
Yes

**Comment by wolftobias** [comment]
Maybe something is missing with the build script, which is generating the public headers.

**Comment by richsalz** [comment]
No, it was human error that left some things out.

**Comment by matthiasballreich** [comment]
an who can fix this now?

**Comment by wolftobias** [comment]
I`ve the same question. :-)

**Comment by richsalz** [comment]
Anyone can.  Make a pull request.

**Comment by wolftobias** [comment]
I needed this feature, therefore I pulled the source code from git. Currently I do not understand how the code for the public headers is generated. 

Everthing I need is in the header crypto/x509v3/v3_admis.h

How can I create the public headers out of it? Is the build script doing this?

**Comment by matthiasballreich** [comment]
@wolftobias take a look [here](https://github.com/openssl/openssl/issues/3788). I made a pull request.
I moved the content from the `v3_admis.h` into the `x509v3.h` and rebuild the `util/libcrypto.num`.

**Comment by wolftobias** [comment]
That sound`s great! Could you bring also the embedded pointers of ADMISSION_SYSNTAX to the public header files?

NAMING_AUTHORITY;
PROFESSION_INFO;
ADMISSIONS;
ADMISSION_SYNTAX;

**Comment by richsalz** [comment]
The code for the public header files is not generated.  Except for error codes.

**Comment by wolftobias** [comment]
ping @matthiasballreich: Are the admission stuff now in the public headers? I tried to find it here in the master, but could`nt find it in x509v3.h header?!

**Comment by matthiasballreich** [comment]
No pull request was not merged. But i saw there was a conflict and fixed it now, so that they can merge it. See [here](https://github.com/openssl/openssl/pull/3822)

**Comment by mattcaswell** [comment]
See #4724.
